### PR TITLE
[257,253] Remove WatchOptions as no longer needed

### PR DESCRIPTION
**Pull request by gaohoward**

### CODE REVIEW

File: controllers/activemqartemis_controller.go [hunk 1]
```diff
@@ -128,33 +126,27 @@ type ActiveMQArtemisReconciler struct {
 func (r *ActiveMQArtemisReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
 	reqLogger := ctrl.Log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name, "Reconciling", "ActiveMQArtemis")
 
+	reqLogger.Info("Reconciling ActiveMQArtemis", request.Name, request.Namespace)
```
Comment (gtully) [inline controllers/activemqartemis_controller.go:129]: maybe remove this, the Reconciling is already in the line above as a nv pair

**Review by gtully — CHANGES_REQUESTED** [review]
Just to clarify... the reconcilers used to have a flag/list of namespaces and filtered events on reconcile.
now the controller has a list of namespaces and the event cache is filtered at source? Events not in a matching namespace never reach the reconcilers now.

File: controllers/activemqartemisaddress_controller.go [hunk 2]
```diff
@@ -78,14 +76,7 @@ type ActiveMQArtemisAddressReconciler struct {
...
 	reqLogger := log.FromContext(ctx).WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name, "Reconciling", "ActiveMQArtemisAddress")
-
-	//remove nsoptions package once migration to new sdk ready
-	//as we use the built-in api in main.go to watch
-	//multiple namespaces
-	if !r.WatchOptions.Match(request.Namespace) {
-		reqLogger.Info("Request not in watch list, ignore", "request", request)
-		return ctrl.Result{}, nil
-	}
+	reqLogger.Info("Reconciling ActiveMQArtemisAddress")
```
[the remark this answers is not in the document]
Comment (gaohoward) [reply controllers/activemqartemisaddress_controller.go:79]: +1, yes I didn't notice the information is added to the logger already.

File: controllers/suite_test.go [hunk 3]
```diff
@@ -118,10 +121,41 @@ func setUpEnvTest() {
...
+func createControllerManager(disableMetrics bool, watchNamespaces ...string) {
+	ctx, cancel = context.WithCancel(context.TODO())
+
+	mgrOptions := ctrl.Options{
 		Scheme: scheme.Scheme,
-	})
+	}
+
+	if len(watchNamespaces) > 1 {
+		logf.Log.Info("##### Set up manager to watch multiple namespaces #####")
```
[the remark this answers is not in the document]
Comment (gaohoward) [reply controllers/suite_test.go:135]: +1

File: main.go [hunk 4]
```diff
@@ -164,7 +158,6 @@ func main() {
 
 	if !watchAllNamespaces && strings.Contains(watchNameSpace, ",") {
 		watchList := strings.Split(watchNameSpace, ",")
-		nsoptions.SetWatchList(watchList)
 		log.Info("Watching multiple namespaces", "value", watchList)
 		//watch all namespace but filter out those not in the list
 		watchNameSpace = ""
```
[the remark this answers is not in the document]
Comment (gaohoward) [reply main.go:163]: +1

**Review by gtully — APPROVED** [review]

**Comment by gaohoward** [comment]
> 

Yes I think that's expected. However I don't expect the event behavior to be changed after we remove the WatchOptions. This is because since we upgraded to the new operator-sdk, both cache and WatchOptions have been used and the WatchOptions is duplicate (which means the *match* call inside a reconciler always returns true). Are there any specific events we are missing?

**Pull request merged**